File: LeetCode/1.TwoSum/Java/alternative_solution/TwoSum2.java
import java.util.Scanner;
import java.util.HashMap;

public class TwoSum2 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the size of the array : ");
        int size = scanner.nextInt();

        int[] array1 = new int[size];

        for (int i = 0;i < size;i++) {
            System.out.printf("Enter the %dth element of the array : ",i + 1);
            array1[i] = scanner.nextInt();
        }

        System.out.print("Enter the target element : ");
        int target = scanner.nextInt();

        int[] result = Twosum(array1,target);   // Method name is not TwoSum, but it is instead Twosum

        if (result == null) {
            System.out.println("No two array elements add upto the target element(" + target + ")");
        } else {
            System.out.printf("Indices of the array elements which add up to %d are : %d and %d \n",target,result[0] + 1,result[1] + 1);
        }

        scanner.close();
    }

    public static int[] Twosum(int[] arr1,int target) {
        HashMap<Integer,Integer> map = new HashMap<>();

        for (int i = 0;i < arr1.length;i++) {
            int complement = target - arr1[i];

            if (map.containsKey(complement)) {
                return new int[]{map.get(complement),i};
            } else {
                map.put(arr1[i],i);
            }
        }

        return null;
    }
}

File: LeetCode/1.TwoSum/C/alternative_solution/TwoSum2.c
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>    // required for map[128]

int* TwoSum(int* arr1,int target);

int main()
{
    int size,target;

    printf("Enter the size of the array : ");
    scanf("%d",&size);

    int* array1 = (int*)malloc(sizeof(int) * size);

    for (int i = 0;i < size;i++)
    {
        printf("Enter the %dth element of the array : ",i + 1);
        scanf("%d",&array1[i]);
    }

    printf("Enter the target element : ");
    scanf("%d",&target);

    int* result = TwoSum(array1,target);

    if (result == NULL)
    {
        printf("No two array elements add up to %d \n",target);
    }
    else
    {
        printf("Indices of the array elements which add up to %d are : %d and %d \n",target,result[0] + 1,result[1] + 1);
    }

    return 0;
}

int* TwoSum(int* arr1,int target)
{
    int* result = (int*)malloc(sizeof(int) * 2);

    int map[128] = {};

    for (int i = 0;i < sizeof(arr1);i++)
    {
        int complement = target - arr1[i];

        if (map[complement])
        {
            result[0] = map[complement];
            result[1] = i;
            return result;
        }
        else
        {
            map[arr1[i]] = i;
        }
    }

    return NULL;
}